style.md

```diff
@@ -1047,6 +1049,62 @@ BenchmarkGood-4  500000000   3.25 ns/op
 </td></tr>
 </tbody></table>
 
+### Prefer Specifying Map Capacity Hints
+
+Where possible, provide capacity hints when initializing
+maps with `make()`.
+
+```go
+make(map[T1]T2, hint)
+```
+
+While Go's runtime map implementation only acts upon capacity
+hints in certain situations (see
+[`runtime.makemap()`](https://golang.org/src/runtime/map.go)
+for more information), providing a hint to `make()` can
+cause allocations to occur at map initialization time rather
+than periodically as elements are added to the map.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+m := make(map[T1]T2)
+
+// later...
+
+for i := 0; i < 10000; i++ {
+  // add element to m
+)
+```
+
+</td><td>
+
+```go
+m := make(map[T1]T2, 10000)
+
+// later...
+
+for i := 0; i < 10000; i++ {
+  // add element to m
+}
+```
+
+</td></tr>
+<tr><td>
+
+Bucket allocations are deferred until elements are added.
+
+</td><td>
+
+Bucket allocations occur at initialization time
+(for sufficiently large capacity hints).
```

Comment:
These may be better as `Less allocations at assignment time` vs `More allocations at assignment time`?